app/models/trip.rb

```diff
@@ -0,0 +1,18 @@
+class Trip < ApplicationRecord
+  before_save :calculate_distance
+  validates_presence_of :start_address, :destination_address
+  validates :price, numericality: { greater_than: 0 }
+
+  private
+
+  def calculate_distance
+    start_point = get_coordinates(start_address)
+    end_point = get_coordinates(destination_address)
+    distance = start_point.distance_from(end_point, :units=>:kms)
+    self.distance = distance.round(2)
```

Comment:
I would prefer to have the database `distance` field be of decimal rather than float and not round the unit. Also prefer to get and store the distance in `meters` rather than `kms`.